src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,362 @@
+            string[] v6connections = tcp6FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            TcpConnectionInformation[] connections = new TcpConnectionInformation[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // UDP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            // UDP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            return connections;
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            string tcp4FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp4ConnectionsFile);
+            string[] v4connections = tcp4FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            IPEndPoint[] endPoints = new IPEndPoint[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // TCP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            // TCP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            return endPoints;
+        }
+
+        public override IPEndPoint[] GetActiveUdpListeners()
+        {
+            string udp4FileContents = File.ReadAllText(LinuxNetworkFiles.Udp4ConnectionsFile);
```

Comment:
> Do you know if there's anything in particular to watch out for here, regarding Encodings and procfs files?

Auto-detect should be fine.  I expect it's possible but unlikely to be anything other than ASCII or UTF8.

> does the above sound accurate?

There are areas of procfs that you need to be a superuser to access, such as certain system processes.  I don't know if /proc/net is one of them, but in general with procfs we should be prepared for UnauthorizedAccessExceptions, as well as FileNotFound/DirectoryNotFoundExceptions for "files" that may come and go based on the underlying state of the system.  For example, in System.Diagnostics.Process, we enumerate the files under /proc/ looking for integers, which are directories that represent each process on the system, but by the time we go to open that directory, the process could have gone away, and the directory will no longer exist.